sylt-compiler/src/typechecker.rs

```diff
@@ -1363,6 +1395,40 @@ impl TypeChecker {
 
                 Constraint::Contains(b) => self.contains(span, ctx, a, *b),
                 Constraint::IsContainedIn(b) => self.contains(span, ctx, *b, a),
+
+                Constraint::Enum => match self.find_type(a) {
+                    Type::Unknown | Type::Enum(..) => Ok(()),
```

Comment:
This seems more like a unify. But it may not be possible to do it any other way

Reply:
I don't want to modify the enum once the type is known - so It's hard to fill in the variant-information if all we know that it's an enum. I didn't want incomplete types in the typesystem, better they are unknowns were my tought.